jib-core/src/main/java/com/google/cloud/tools/jib/builder/NonBlockingListenableFuture.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ListenableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+
+/** A {@link ListenableFuture} decorator that makes sure that {@link #get} is only called after the future has returned, and thus never blocks a thread. */
+class NonBlockingListenableFuture<T> implements ListenableFuture<T> {
```

Comment:
This class does seem a little odd (Since it extends listenableFuture and also holds inside it a listenable future) and kinda makes me think we might not being doing this right. 

Perhaps `isGetEnabled` can satisfied by `isDone`?

Reply:
Changed to just one static method called `NonBlockingFutures#get`